pkg/manifest/mantaray/marshal.go

```diff
@@ -270,13 +270,6 @@ func (n *Node) UnmarshalBinary(data []byte) error {
 
 		n.entry = append([]byte{}, data[nodeHeaderSize:nodeHeaderSize+refBytesSize]...)
 		offset := nodeHeaderSize + refBytesSize // skip entry
-		// Currently we don't persist the root nodeType when we marshal the manifest, as a result
-		// the root nodeType information is lost on Unmarshal. This causes issues when we want to
-		// perform a path 'Walk' on the root. If there is more than 1 fork, the root node type
-		// is an edge, so we will deduce this information from index byte array
-		if !bytes.Equal(data[offset:offset+32], make([]byte, 32)) {
-			n.nodeType = nodeTypeEdge
-		}
```

Comment:
IMO this edge case handling could remain here, just instead of overwriting the `n.nodeType`, the `makeEdgeType` method of `n` should be called, so `n.nodeType = nodeTypeEdge` -> `n.makeEdge()`

Reply:
true, plus actually an all zeros check does not new allocation each time, just compare to `swarm.ZeroAddress.Bytes()` or do it with a loop

Reply:
Agreed. I have actually added tests now to verify the behavior. Most of the bugs we saw were after serialization and deserialization and the tests don't test this behavior.

For all the relevant tests I have added a `with load store` variant which will do the serialization-deserialization and re-run the tests.

@zelig We don't have the notion of `swarm.Address` in the mantaray pkg. Instead, I have created a new constant that is local to this pkg.